Remove case sensitivity for variable dimensions

peverwhee opened the pull request:
Updates metadata_table.py parsing to lowercase the following fields:
* local name
* standard name
* dimensions

Context: I ran into an issue where dimension variable standard names were not being found if they contain >=1 upper case letter.

To reproduce, modify any dimension in any of the capgen tests to be mixed case. The test run will fail with this message: `Could not find dimension...`

User interface changes?: No

Commit:
grab dimensions in lower case

climbfuji reviewed (commented):
Is this the only place where a lowercase version of dimension names is needed? Or should this be done much deeper down in capgen when parsing the metadata files?

gold2718 reviewed (approved):
I'm okay with this if it solves whatever problem caused this work (there is no issue or problem mentioned in the PR description). It is a standard accessing routine.

peverwhee commented:
@gold2718 - I added a bit more context to the PR description!

@climbfuji - a good question. I'll look into how tricky it is to `lower` all of the variable standard names and dimensions at parse time.

Commit:
new approach to case agnostic dimensions

peverwhee commented:
@climbfuji @gold2718 

Dom was right to suggest an "earlier" change to lowercase - there was at least one other path through the code that resulted in the same error. I updated the parser to return both dimensions and standard names in lower case.

One question for y'all and for @dustinswales :

Thanks to Dustin's recent PR, numerical dimensions are OK (like `(horizontal_dimension, 2)`) but what about numerical indices for dimensions (like `(2:horizontal_dimension)`)? Is that allowed? If so, it's not working and I'll open an issue.

climbfuji commented:
> @climbfuji @gold2718
> 
> Dom was right to suggest an "earlier" change to lowercase - there was at least one other path through the code that resulted in the same error. I updated the parser to return both dimensions and standard names in lower case.
> 
> One question for y'all and for @dustinswales :
> 
> Thanks to Dustin's recent PR, numerical dimensions are OK (like `(horizontal_dimension, 2)`) but what about numerical indices for dimensions (like `(2:horizontal_dimension)`)? Is that allowed? If so, it's not working and I'll open an issue.

I think that `(2:horizontal_dimension)` is allowed.

gold2718 reviewed (approved):
Still looks okay to me.

dustinswales commented:
> > @climbfuji @gold2718
> > Dom was right to suggest an "earlier" change to lowercase - there was at least one other path through the code that resulted in the same error. I updated the parser to return both dimensions and standard names in lower case.
> > One question for y'all and for @dustinswales :
> > Thanks to Dustin's recent PR, numerical dimensions are OK (like `(horizontal_dimension, 2)`) but what about numerical indices for dimensions (like `(2:horizontal_dimension)`)? Is that allowed? If so, it's not working and I'll open an issue.
> 
> I think that `(2:horizontal_dimension)` is allowed.

I concur.

dustinswales reviewed (approved):
There changes look good to me.
One question. I wonder if we still need [this change I made ](https://github.com/NCAR/ccpp-framework/pull/652/files#diff-6c29fef823c4374ccb84a501792221eedede638bec1d6f9c4c3b40c3252f4c64R711) with your changes to the parser?

Commit:
lowercase local name during parsing as well

peverwhee commented:
@dustinswales 

Good question! I removed that line from your previous PR and also added a line in the parser to make the local names lowercase as well.
